pytorch_lightning/core/hooks.py

```diff
@@ -594,6 +606,18 @@ def predict_dataloader(self) -> Union[DataLoader, List[DataLoader]]:
             will have an argument ``dataloader_idx`` which matches the order here.
         """
 
+    def on_request_train_dataloader(self) -> None:
+        """Called before requesting the train dataloader."""
+
+    def on_request_val_dataloader(self) -> None:
+        """Called before requesting the val dataloader."""
+
+    def on_request_test_dataloader(self) -> None:
+        """Called before requesting the test dataloader."""
+
+    def on_request_predict_dataloader(self) -> None:
+        """Called before requesting the predict dataloader."""
+
```

Comment:
n00b q: how are these hooks intended to be used?

Reply:
In Flash, those hooks are used to inject the attach the data_pipeline into the model dataloaders before they are being fetched.

Reply:
can't this just become `on_train_dataloader` etc?

Reply:
Yes, done !